python/pdstools/app/health_check/pages/3_Reports.py

```diff
@@ -101,22 +99,27 @@
             )
 
     except Exception as e:
-        st.error(f"""An error occured: {e}""")
-        traceback_str = traceback.format_exc()
-        with open(working_dir / "log.txt", "a") as f:
-            f.write(traceback_str)
-        with open(working_dir / "log.txt", "rb") as f:
-            btn = st.download_button(
-                label="Download error log",
-                data=f,
-                file_name="errorlog.txt",
-                key="ErrorLogDownload",
-            )
+        if "HCErrorLogDownload" not in st.session_state:
+            st.error(f"""An error occurred hc: {e}""")
+            traceback_str = traceback.format_exc()
+
+            # Find the most recent log file in the working directorpdsy
+            log_files = list(working_dir.glob("pdstools_logs_*.txt"))
```

Comment:
Shall we keep this user flexibility to set working_dir? I don't see how this can be dangerous.
![Screenshot 2024-08-22 at 17 59 11](https://github.com/user-attachments/assets/46d554dd-d43c-424f-9833-188a14a0e8df)

Reply:
I agree - but I suppose since you're calling a glob on top of a working_dir variable, it may lead to variance. Maybe it would be more reliable to put the working_dir in an f-string?